### PR TITLE
fix bug #12122

### PR DESCRIPTION
**Pull request by zhaozg**

### CODE REVIEW

File: runtime/lua/vim/lsp/util.lua [hunk 1]
```diff
@@ -725,6 +725,7 @@ do
         [protocol.DocumentHighlightKind.Read] = "LspReferenceRead";
         [protocol.DocumentHighlightKind.Write] = "LspReferenceWrite";
       }
+      reference["kind"] = reference["kind"] or protocol.DocumentHighlightKind.Text
```
Comment (teto) [inline runtime/lua/vim/lsp/util.lua:728]: best not to override a variable (ease debug). Use a new one like `local kind = reference["kind"] or protocol.DocumentHighlightKind.Text`

**Comment by teto** [comment]
thanks for the patch, please make the commit message more descriptive such as
```
lsp: provide a default for missing reference kind

Fix #XXXXX
<same description as in your issue>
```

**Comment by zhaozg** [comment]
> thanks for the patch, please make the commit message more descriptive such as
> 
> ```
> lsp: provide a default for missing reference kind
> 
> Fix #XXXXX
> <same description as in your issue>
> ```

Follow your recommand

**Comment by teto** [comment]
thanks for editing the commit message, it's good. Could you address https://github.com/neovim/neovim/pull/12127#discussion_r409514843 and I will merge. Thanks

**Pull request merged**